docs/DeveloperGuide.md

```diff
@@ -1,33 +1,69 @@
 # Developer Guide
```

Comment:
wow thank u for adding the table of contents!! would be useful for navigating the DG